File: src/main/java/org/aldo/api/data/dto/MonthlyDetailDto.java
package org.aldo.api.data.dto;

import lombok.Data;

import java.util.List;

@Data
public class MonthlyDetailDto {
    private SummaryProfessorDto professor;
    private YearlyHoursDto collaborationHoursYearly;
    private List<MonthlyHoursDto> collaborationHoursMonthly;
}

File: src/app/model/dto/MonthlyDetailDto.ts
import {SummaryProfessorDto} from "./SummaryProfessorDto";
import {YearlyHoursDto} from "./YearlyHoursDto";
import {MonthlyHoursDto} from "./MonthlyHoursDto";

export class MonthlyDetailDto {
  professor: SummaryProfessorDto;
  collaborationHoursYearly: YearlyHoursDto;
  collaborationHoursMonthly: MonthlyHoursDto[];

  constructor(data: any) {
    this.professor = new SummaryProfessorDto(data.professor);
    this.collaborationHoursYearly = data.collaborationHoursYearly;
    this.collaborationHoursMonthly = data.collaborationHoursMonthly;
  }
}
